frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -0,0 +1,70 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.ui.DocumentAdapter
+import com.intellij.ui.layout.CellBuilder
+import com.intellij.ui.layout.ValidationInfoBuilder
+import com.intellij.ui.layout.panel
+import com.virtuslab.gitmachete.frontend.actions.common.GitMacheteBundle.message
+import kotlin.apply
+import kotlin.text.isEmpty
+import kotlin.text.trim
+
+class SlideInDialog constructor(project: Project, private val parentName: String) :
+    DialogWrapper(project, true) {
+
+  // this field is only ever meant to be written on UI thread
+  private var branchName = ""
+
+  init {
+    title = message("action.GitMachete.BaseSlideInBranchBelowAction.dialog.slide-in.title")
+    setOKButtonText(
+        message("action.GitMachete.BaseSlideInBranchBelowAction.dialog.slide-in.ok-button"))
+    setOKButtonMnemonic('I'.toInt())
+    init()
+  }
+
+  fun showAndGetBranchName() = if (showAndGet()) branchName.trim() else null
+
+  override fun createCenterPanel() =
+      panel {
+        row(
+            message(
```

Comment:
Nit: increase max line length? if easily doable

Reply:
the funny thing is that spotless does not allow that 

`not easily doable`